testutils/src/main/scala/za/co/absa/enceladus/testutils/DataframeReader.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.testutils
+
+import org.apache.spark.sql.types.{StructField, StructType}
+import org.apache.spark.sql.{DataFrame, DataFrameReader, SparkSession}
+
+import scala.annotation.switch
+
+class DataframeReader(datasetPath: String, inputSchema: StructType = null)(implicit cmd: CmdConfig, sparkSession: SparkSession) {
+  lazy val dataFrame: DataFrame = getDataFrameReader.load(datasetPath)
+  lazy val dataFrameSchema: StructType = dataFrame.schema
+
+  def removeMetadataFromSchema: StructType = {
+    StructType(dataFrameSchema.map{ f => StructField(f.name, f.dataType, f.nullable) })
+  }
+
+  private def getDataFrameReader(implicit cmd: CmdConfig, sparkSession: SparkSession): DataFrameReader = {
+    (cmd.rawFormat: @switch) match {
```

Comment:
This is interesting, I've never seen this before! Had to check it out, apparently it provides a warning if the `match` does not compile into a [tableswitch or lookupswitch](https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-3.html#jvms-3.10).

Reply:
It shouldn't make difference to the bytecode but could pop a warning if we misuse the match.

Reply:
Yeah, I was just surprised. I'm not sure about how useful it will be, but let's give it a try.